modules/json/src/smithy4s/http/json/codecs.scala

```diff
@@ -43,13 +44,18 @@ object codecs {
       // TODO: add tests for `codecs` understanding int enums. Maybe pizza spec
       IntEnum
     )
+  val defaultMaxArity: Int = 1024
 
   private[smithy4s] def schemaVisitorJCodec(
-      cache: CompilationCache[JCodec]
+      cache: CompilationCache[JCodec],
+      maxArity: Int = defaultMaxArity
   ): SchemaVisitor[JCodec] =
-    new SchemaVisitorJCodec(maxArity = 1024, cache)
+    new SchemaVisitorJCodec(maxArity, cache)
 
   private[smithy4s] val schemaVisitorJCodec: SchemaVisitor[JCodec] =
-    new SchemaVisitorJCodec(maxArity = 1024, CompilationCache.nop[JCodec])
+    new SchemaVisitorJCodec(
+      maxArity = defaultMaxArity,
+      CompilationCache.nop[JCodec]
+    )
```

Comment:
What is this version currently used for? From a quick look it seems like only in tests. Maybe we can remove.

Reply:
Im not sure what you mean by version?

Reply:
> Let's add some documentation showing how to increase the maxArity and what that means

good point

Reply:
> Im not sure what you mean by version?

Yeah sorry that is super confusing! I meant the "Val" vs the "def" above. The val I'm not sure is used anywhere. Thought we could remove it if that is in fact the case.